Update zh-cn translation related to color and gradient (#14819)
mdn/translated-content: files/zh-cn/web/css/color_value/index.md

{{CSSRef}}

[CSS](/zh-CN/docs/Web/CSS) [数据类型](/zh-CN/docs/Web/CSS/CSS_Types) **`<color>`** 表示一个颜色。颜色可以包括一个 [alpha 通道](https://zh.wikipedia.org/wiki/Alpha合成)_透明度值_，来表明颜色如何与背景色[混合（composite）](http://www.w3.org/TR/2003/REC-SVG11-20030114/masking.html#SimpleAlphaBlending)。

> **备注：** 尽管 CSS 颜色值（`<color>`）被精确定义，但在不同的输出设备上仍然有可能表现不一致（甚至区别很大）。这是因为大多数设备不是经校准的，而且有些浏览器不支持输出设备的[色彩配置（color profile）](https://zh.wikipedia.org/wiki/ICC色彩特性文件)。

## 语法

```css

> /* 命名的颜色 */

Reviewer: 与named-color那边到时统一下

```css
/* 这两个是等价的 */
color: oklab(50% none -0.25);
color: oklab(50% 0 -0.25);

/* 这两个是等价的 */
background-color: hsl(none 100% 50%);
background-color: hsl(0deg 100% 50%);
```

## 插值

在动画和[渐变](/zh-CN/docs/Web/SVG/Tutorial/Gradients)中， `<color>` 值会在每个红绿蓝成分中插值。每一个成分都会插值为浮点实数。注意颜色的插值发生在 [alpha-premultiplied sRGBA color space](https://www.gimp.org/docs/plug-in/appendix-alpha.html) 中，以避免出现未预料的灰色。在动画中，插值速度由[时间函数](/zh-CN/docs/)决定。

颜色插值会发生在[渐变](/zh-CN/docs/Web/CSS/gradient)、[过渡](/zh-CN/docs/Web/CSS/CSS_transitions/Using_CSS_transitions) 和 [动画](/zh-CN/docs/Web/CSS/CSS_animations/Using_CSS_animations) 中。

插值 `<color>` 值时，首先会转换成指定的颜色空间，然后颜色成分的[计算值](/zh-CN/docs/Web/CSS/computed_value)线性插值，插值速度由过渡和动画中的[缓动函数](/zh-CN/docs/Web/CSS/easing-function)决定。插值颜色空间默认为 Oklab，但是可以在一些与颜色相关的函数标记中通过 {{CSSXref("&lt;color-interpolation-method&gt;")}} 覆盖。

### 带有缺失成分时的插值

#### 在同一空间内插值颜色

对完全在插值颜色空间中的颜色进行插值时，一个颜色中缺失的成分将替换为另一种颜色中同一成分的现有值。例如，以下两个表达式是等效的：

```css
color-mix(in oklch, oklch(none 0.2 10), oklch(60% none 30))
color-mix(in oklch, oklch(60% 0.2 10), oklch(60% 0.2 30))
```

> **备注：** 如果两个颜色都缺失这个成分，那么这个成分在插值后也是缺失的。

#### 在不同空间内插值：相似成分

如果任何要插值的颜色不在插值颜色空间中，则其缺失的成分将根据同一类的**相似成分**转换为转换后的颜色，如下表所述：

> | 类似 | 相似成分 |

Reviewer: ```suggestion
| 类别 | 相似成分 |
```

> 
> 在动画和[渐变](/zh-CN/docs/Web/SVG/Tutorial/Gradients)中， `<color>` 值会在每个红绿蓝成分中插值。每一个成分都会插值为浮点实数。注意颜色的插值发生在 [alpha-premultiplied sRGBA color space](https://www.gimp.org/docs/plug-in/appendix-alpha.html) 中，以避免出现未预料的灰色。在动画中，插值速度由[时间函数](/zh-CN/docs/)决定。

Reviewer: ```suggestion
```

| ---- | -------- |
| 红色 | `R`, `X` |
| 绿色 | `G`, `Y` |
| 蓝色 | `B`, `Z` |
| 明度 | `L`      |
| 色度 | `C`, `S` |
| 色相 | `H`      |
| a    | `a`      |
| b    | `b`      |

例如：

- `color(xyz 0.2 0.1 0.6)` 中的 `X`（`0.2`）相似于 `rgb(50% 70% 30%)` 中的 `R`（`50%`）。
- `hsl(0deg 100% 80%)` 中的 `H`（`0deg`）相似于 `oklch(80% 0.1 140)` 中的 `H`（`140`）。

使用 Oklch 作为插值颜色空间并以下面两种颜色为例：

```css
lch(80% 30 none)
color(display-p3 0.7 0.5 none)
```

预处理过程为：

1. 将两种颜色中缺失的成分替换为零值：

   ```css
   lch(80% 30 0)
   color(display-p3 0.7 0.5 0)
   ```

2. 将两种颜色转换为插值颜色空间：

   ```css
   oklch(83.915% 0.0902 0.28)
   oklch(63.612% 0.1522 78.748)
   ```

3. 如果转换后颜色的任何成分与相应原始颜色中的缺失成分相似，则将其重置为缺失的成分：

   ```css
   oklch(83.915% 0.0902 none)
   oklch(63.612% 0.1522 78.748)
   ```

4. 将任何缺失的成分替换为另一转换后颜色中的相同成分：

   ```css
   oklch(83.915% 0.0902 78.748)
   oklch(63.612% 0.1522 78.748)
   ```

## 无障碍考虑

> 一些人分辨颜色可能会存在困难。[WCAG 2.2](/zh-CN/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Use_of_color) 强烈建议不要将颜色作为传递特定消息、活动或结果的唯一方式。参见[颜色和颜色对比度](/zh-CN/docs/Learn/Accessibility/CSS_and_JavaScript#颜色对比度)以了解更多信息。

Reviewer: ```suggestion
一些人分辨颜色可能会存在困难。[WCAG 2.2](/zh-CN/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Use_of_color) 强烈建议不要将颜色作为传递特定消息、活动或结果的唯一方式。参见[颜色和颜色对比度](/zh-CN/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast)以了解更多信息。
```

> 一些人分辨颜色可能会存在困难。[WCAG 2.0](https://www.w3.org/TR/WCAG/#visual-audio-contrast) 强烈建议不要将颜色作为传递特定消息、活动或结果的唯一方式。参见 [颜色和颜色对比度](/zh-CN/docs/Learn/Accessibility/CSS_and_JavaScript#Color_and_color_contrast) 以了解更多信息。

Reviewer: ```suggestion
一些人分辨颜色可能会存在困难。[WCAG 2.2](/zh-CN/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Use_of_color) 强烈建议不要将颜色作为传递特定消息、活动或结果的唯一方式。参见[颜色和颜色对比度](/zh-CN/docs/Learn/Accessibility/CSS_and_JavaScript#颜色对比度)以了解更多信息。
```

## 参见

- {{CSSXref("opacity")}}：在元素级别上设备透明度的属性
- {{CSSXref("&lt;hue&gt;")}}：表示一个颜色的色相角度的数据类型
- {{CSSXref("color")}}、{{CSSXref("background-color")}}、{{CSSXref("border-color")}}、{{CSSXref("box-shadow")}}、{{CSSXref("outline-color")}}、{{CSSXref("text-shadow")}}：使用 `<color>` 的常见属性
- [使用 CSS 给元素应用颜色](/zh-CN/docs/Web/CSS/CSS_colors/Applying_color)

> - MDN blog（2023）中的 [CSS 颜色中的新功能、渐变和色调（级别 4）](/en-US/blog/css-color-module-level-4/)

Reviewer: ```suggestion
- MDN 博客中的文章 [CSS 颜色中的新功能、渐变和色调（级别 4）](/en-US/blog/css-color-module-level-4/)（2023 年）
```

## 参见

- {{CSSXref("opacity")}}：在元素级别上设备透明度的属性
- {{CSSXref("&lt;hue&gt;")}}：表示一个颜色的色相角度的数据类型
- {{CSSXref("color")}}、{{CSSXref("background-color")}}、{{CSSXref("border-color")}}、{{CSSXref("box-shadow")}}、{{CSSXref("outline-color")}}、{{CSSXref("text-shadow")}}：使用 `<color>` 的常见属性
- [使用 CSS 给元素应用颜色](/zh-CN/docs/Web/CSS/CSS_colors/Applying_color)

> - MDN blog（2023）中的[新函数、渐变、CSS 颜色中的色相（级别 4）](/zh-CN/blog/css-color-module-level-4/)

Reviewer: 博客是暂时没有本地化链接的，目前这里的红链还是误报状态，待yari平台修复

相关issue：https://github.com/mdn/yari/issues/9345

```suggestion
- MDN blog（2023）中的[新函数、渐变、CSS 颜色中的色相（级别 4）](/en-US/blog/css-color-module-level-4/)
```

> - MDN blog（2023）中的[新函数、渐变、CSS 颜色中的色相（级别 4）](/en-US/blog/css-color-module-level-4/)

Reviewer: ```suggestion
- MDN blog（2023）中的 [CSS 颜色中的新功能、渐变和色调（级别 4）](/en-US/blog/css-color-module-level-4/)
```
